scripts/data_files/driver_jsons/driver_opaque_schema.json

```diff
@@ -0,0 +1,112 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "properties": {
+    "prefix": {
+      "type": "string"
+    },
+    "type": {
+      "type": "string"
+    },
+    "location": {
+      "type": "string"
```

Comment:
Do you mean anyOf { "type": "string", "type": "int" }
Why int, A pattern validated string should suffice. We will be directly templating the validated string. I can appropriately update the spec.

Reply:
Isn't `"anyOf" { "type": "string", "type": "int" }` equivalent to `"type": ["string", "int"]`?

Why not allow an integer here? In general, this can be any C constant expression that evaluates to an integer. This includes things like `"123"`. Why not allow `123` as well? The driver specification currently states that C-integer-valued properties can be expressed as either integers or strings in JSON.

Reply:
> As per my compiler, NOT.

I can't figure out the difference by reading the JSON schema specification. Can you please give an example of a difference? If you don't know either, what compiler are you using and what exactly does it say about the difference?

----

> Since the expanded macro, say "MY_DRIVER_LOCATION" would be defined in the header file and can be used by applications using the same header file.

Yes, it is definitely a goal to allow this. And yes, it means that uniqueness can only be tested in C, not syntactically.

Reply:
https://cswr.github.io/JsonSchema/spec/multiple_types/
Multiple_types are allowed, the error thrown by my compiler was quite cryptic, it wasn't on type taking multiple options but rather supporting _integer_ instead of _int_. Fixed !

anyOf is still an option to give more granular tuning to each type.